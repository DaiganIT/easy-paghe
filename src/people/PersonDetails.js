import React from 'react';
import { withStyles, Paper, Typography, TextField, Grid } from '@material-ui/core';

const styles = {
  paper: {
    padding: '1em',
		marginBottom: '1em',
  },
  miniTitle: {
    marginBottom: '1em',
  },
  textField: {
    marginRight: '1em',
		marginBottom: '1em',
  }
}

function PersonDetails({ classes, isSaving, person, updateField, errors }) {
	return (
		<React.Fragment>
			<Grid container spacing={16}>
				<Grid item xs={12} md={6}>
					<TextField
						id="person-first-name"
						error={!!errors.firstName}
						required
						variant="outlined"
						label="Nome"
						fullWidth
						className={classes.textField}
						disabled={isSaving}
						value={person.firstName}
						onChange={(e) => updateField('firstName', e.target.value)}
					/>
				</Grid>
				<Grid item xs={12} md={6}>
					<TextField
						id="person-last-name"
						error={!!errors.lastName}
						required
						variant="outlined"
						label="Cognome"
						fullWidth
						className={classes.textField}
						disabled={isSaving}
						value={person.lastName}
						onChange={(e) => updateField('lastName', e.target.value)}
					/>
				</Grid>
				<Grid item xs={12}>
					<TextField
						id="person-address"
						error={!!errors.address}
						variant="outlined"
						label="Indirizzo"
						fullWidth
						className={classes.textField}
						disabled={isSaving}
						value={person.address}
						onChange={(e) => updateField('address', e.target.value)}
					/>
				</Grid>
				<Grid item xs={12} md={6}>
					<TextField
						id="person-phone"
						error={!!errors.phone}
						variant="outlined"
						label="Telefono"
						fullWidth
						className={classes.textField}
						disabled={isSaving}
						value={person.phone}
						onChange={(e) => updateField('phone', e.target.value)}
					/>
				</Grid>
				<Grid item xs={12} md={6}>
					<TextField
						id="person-email"
						error={!!errors.email}
						variant="outlined"
						label="Email"
						fullWidth
						className={classes.textField}
						disabled={isSaving}
						value={person.email}
						onChange={(e) => updateField('email', e.target.value)}
					/>
				</Grid>
			</Grid>
		</React.Fragment>
	);
}

export default withStyles(styles)(PersonDetails);