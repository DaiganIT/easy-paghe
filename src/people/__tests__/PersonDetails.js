import React from 'react';
import {
  render,
  cleanup,
  fireEvent,
} from 'react-testing-library';
import 'jest-dom/extend-expect';
import PersonDetails from '../PersonDetails';

afterEach(cleanup)

const defaultProps = {
  isSaving: false,
  person: {
    firstName: 'Pietro',
    lastName: 'Carta',
    phone: '65654',
    address: 'inkerman road',
    email: 'test@test.it',
  },
  updateField: jest.fn(),
  errors: {}
}

test('Company Details structure snapshot is correct', () => {
  const { container } = render(<PersonDetails {...defaultProps} />);
  expect(container.firstChild).toMatchSnapshot();
});

const testCases = [
  {
    name: 'firstName',
    inputId: 'first-name'
  },
  {
    name: 'lastName',
    inputId: 'last-name'
  },
  {
    name: 'phone',
    inputId: 'phone',
  },
  {
    name: 'address',
    inputId: 'address',
  },
  {
    name: 'email',
    inputId: 'email',
  },
]

testCases.forEach(testCase => {
  afterEach(jest.clearAllMocks);

  test(`Person Details ${testCase.name} input is rendered`, () => {
    const { container } = render(<PersonDetails {...defaultProps} />);

    expect(container.querySelector(`input#person-${testCase.inputId}`).value).toBe(defaultProps.person[testCase.name]);
  });

  test(`Person Details ${testCase.name} input change is catched`, () => {
    const { container } = render(<PersonDetails {...defaultProps} />);

    fireEvent.change(container.querySelector(`input#person-${testCase.inputId}`), { target: { value: 'Changed' } });
    expect(defaultProps.updateField).toHaveBeenCalledWith(testCase.name, 'Changed');
  });
});

test('Person Details inputs are disabled when isSaving is true', () => {
  const props = Object.assign({}, defaultProps, { isSaving: true });
  const { container } = render(<PersonDetails {...props} />);

  const allInputs = container.querySelectorAll('input');
  for(const input of allInputs) {
    expect(input).toBeDisabled();
  }
});

testCases.forEach(testCase => {
  test(`Person Details ${testCase.name} input is rendered with errors`, () => {
    const props = Object.assign({}, defaultProps, { errors: { [testCase.name]: ['error'] }});
    const { container } = render(<PersonDetails {...props} />);

    expect(container.querySelector(`label[for=person-${testCase.inputId}]`).className).toContain('MuiFormLabel-error');
  });
});