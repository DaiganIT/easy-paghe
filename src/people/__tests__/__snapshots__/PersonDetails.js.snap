// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company Details structure snapshot is correct 1`] = `
<div
  class="MuiGrid-container-4 MuiGrid-spacing-xs-16-27"
>
  <div
    class="MuiGrid-item-5 MuiGrid-grid-xs-12-44 MuiGrid-grid-md-6-66"
  >
    <div
      class="MuiFormControl-root-101 MuiFormControl-fullWidth-104 PersonDetails-textField-3"
    >
      <label
        class="MuiFormLabel-root-116 MuiFormLabel-filled-120 MuiFormLabel-required-121 MuiInputLabel-required-109 MuiInputLabel-root-105 MuiInputLabel-formControl-110 MuiInputLabel-animated-113 MuiInputLabel-shrink-112 MuiInputLabel-outlined-115"
        data-shrink="true"
        for="person-first-name"
      >
        Nome
        <span
          class="MuiFormLabel-asterisk-122"
        >
           *
        </span>
      </label>
      <div
        class="MuiInputBase-root-136 MuiOutlinedInput-root-123 MuiInputBase-fullWidth-145 MuiInputBase-formControl-137"
      >
        <fieldset
          aria-hidden="true"
          class="MuiPrivateNotchedOutline-root-153 MuiOutlinedInput-notchedOutline-130"
          style="padding-left: 8px;"
        >
          <legend
            class="MuiPrivateNotchedOutline-legend-154"
            style="width: 0px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
        <input
          aria-invalid="false"
          class="MuiInputBase-input-146 MuiOutlinedInput-input-131"
          id="person-first-name"
          required=""
          type="text"
          value="Pietro"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-5 MuiGrid-grid-xs-12-44 MuiGrid-grid-md-6-66"
  >
    <div
      class="MuiFormControl-root-101 MuiFormControl-fullWidth-104 PersonDetails-textField-3"
    >
      <label
        class="MuiFormLabel-root-116 MuiFormLabel-filled-120 MuiFormLabel-required-121 MuiInputLabel-required-109 MuiInputLabel-root-105 MuiInputLabel-formControl-110 MuiInputLabel-animated-113 MuiInputLabel-shrink-112 MuiInputLabel-outlined-115"
        data-shrink="true"
        for="person-last-name"
      >
        Cognome
        <span
          class="MuiFormLabel-asterisk-122"
        >
           *
        </span>
      </label>
      <div
        class="MuiInputBase-root-136 MuiOutlinedInput-root-123 MuiInputBase-fullWidth-145 MuiInputBase-formControl-137"
      >
        <fieldset
          aria-hidden="true"
          class="MuiPrivateNotchedOutline-root-153 MuiOutlinedInput-notchedOutline-130"
          style="padding-left: 8px;"
        >
          <legend
            class="MuiPrivateNotchedOutline-legend-154"
            style="width: 0px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
        <input
          aria-invalid="false"
          class="MuiInputBase-input-146 MuiOutlinedInput-input-131"
          id="person-last-name"
          required=""
          type="text"
          value="Carta"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-5 MuiGrid-grid-xs-12-44"
  >
    <div
      class="MuiFormControl-root-101 MuiFormControl-fullWidth-104 PersonDetails-textField-3"
    >
      <label
        class="MuiFormLabel-root-116 MuiFormLabel-filled-120 MuiInputLabel-root-105 MuiInputLabel-formControl-110 MuiInputLabel-animated-113 MuiInputLabel-shrink-112 MuiInputLabel-outlined-115"
        data-shrink="true"
        for="person-address"
      >
        Indirizzo
      </label>
      <div
        class="MuiInputBase-root-136 MuiOutlinedInput-root-123 MuiInputBase-fullWidth-145 MuiInputBase-formControl-137"
      >
        <fieldset
          aria-hidden="true"
          class="MuiPrivateNotchedOutline-root-153 MuiOutlinedInput-notchedOutline-130"
          style="padding-left: 8px;"
        >
          <legend
            class="MuiPrivateNotchedOutline-legend-154"
            style="width: 0px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
        <input
          aria-invalid="false"
          class="MuiInputBase-input-146 MuiOutlinedInput-input-131"
          id="person-address"
          type="text"
          value="inkerman road"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-5 MuiGrid-grid-xs-12-44 MuiGrid-grid-md-6-66"
  >
    <div
      class="MuiFormControl-root-101 MuiFormControl-fullWidth-104 PersonDetails-textField-3"
    >
      <label
        class="MuiFormLabel-root-116 MuiFormLabel-filled-120 MuiInputLabel-root-105 MuiInputLabel-formControl-110 MuiInputLabel-animated-113 MuiInputLabel-shrink-112 MuiInputLabel-outlined-115"
        data-shrink="true"
        for="person-phone"
      >
        Telefono
      </label>
      <div
        class="MuiInputBase-root-136 MuiOutlinedInput-root-123 MuiInputBase-fullWidth-145 MuiInputBase-formControl-137"
      >
        <fieldset
          aria-hidden="true"
          class="MuiPrivateNotchedOutline-root-153 MuiOutlinedInput-notchedOutline-130"
          style="padding-left: 8px;"
        >
          <legend
            class="MuiPrivateNotchedOutline-legend-154"
            style="width: 0px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
        <input
          aria-invalid="false"
          class="MuiInputBase-input-146 MuiOutlinedInput-input-131"
          id="person-phone"
          type="text"
          value="65654"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-5 MuiGrid-grid-xs-12-44 MuiGrid-grid-md-6-66"
  >
    <div
      class="MuiFormControl-root-101 MuiFormControl-fullWidth-104 PersonDetails-textField-3"
    >
      <label
        class="MuiFormLabel-root-116 MuiFormLabel-filled-120 MuiInputLabel-root-105 MuiInputLabel-formControl-110 MuiInputLabel-animated-113 MuiInputLabel-shrink-112 MuiInputLabel-outlined-115"
        data-shrink="true"
        for="person-email"
      >
        Email
      </label>
      <div
        class="MuiInputBase-root-136 MuiOutlinedInput-root-123 MuiInputBase-fullWidth-145 MuiInputBase-formControl-137"
      >
        <fieldset
          aria-hidden="true"
          class="MuiPrivateNotchedOutline-root-153 MuiOutlinedInput-notchedOutline-130"
          style="padding-left: 8px;"
        >
          <legend
            class="MuiPrivateNotchedOutline-legend-154"
            style="width: 0px;"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
        <input
          aria-invalid="false"
          class="MuiInputBase-input-146 MuiOutlinedInput-input-131"
          id="person-email"
          type="text"
          value="test@test.it"
        />
      </div>
    </div>
  </div>
</div>
`;
